cg/store/crud/read.py

```diff
@@ -1608,27 +1614,47 @@ def get_related_samples(
                 SampleFilter.BY_CUSTOMER_ENTRY_IDS,
             ],
         )
+        return samples
 
-        return sample_application_version_query.all()
+    def get_uploaded_related_dna_cases(self, rna_case_id: str) -> list[Case]:
+        """Returns all uploaded DNA cases ids related to the given RNA case."""
 
-    def get_related_cases(
-        self, sample_internal_id: str, workflows: list[Workflow], collaborators: set[Customer]
-    ) -> list[Case]:
-        """Return a list of cases linked to the given sample within the given list of workflows and customers in a collaboration."""
+        rna_case: Case = self.get_case_by_internal_id(internal_id=rna_case_id)
 
-        cases_with_samples: Query = self._join_sample_and_case()
-        cases_with_samples: Query = apply_case_sample_filter(
-            case_samples=cases_with_samples,
-            sample_internal_id=sample_internal_id,
-            filter_functions=[CaseSampleFilter.CASES_WITH_SAMPLE_BY_INTERNAL_ID],
-        )
+        related_dna_cases: list[Case] = []
+        for rna_sample in rna_case.samples:
 
-        return apply_case_filter(
-            cases=cases_with_samples,
-            workflows=workflows,
-            customer_entry_ids=[customer.id for customer in collaborators],
-            filter_functions=[
-                CaseFilter.BY_WORKFLOWS,
-                CaseFilter.BY_CUSTOMER_ENTRY_IDS,
-            ],
-        ).all()
+            collaborators: set[Customer] = rna_sample.customer.collaborators
+
+            related_dna_samples: Query = self._get_related_samples_query(
+                sample=rna_sample,
+                prep_categories=DNA_PREP_CATEGORIES,
+                collaborators=collaborators,
+            )
+
+            dna_samples_cases_query = related_dna_samples.join(
+                CaseSample, Case.id == CaseSample.case_id
+            ).join(Case, CaseSample.case_id == Case.id)
+
+            dna_samples_cases_query: Query = apply_case_filter(
+                cases=dna_samples_cases_query,
+                workflows=DNA_WORKFLOWS_WITH_SCOUT_UPLOAD,
+                customer_entry_ids=[customer.id for customer in collaborators],
+                filter_functions=[
+                    CaseFilter.BY_WORKFLOWS,
+                    CaseFilter.BY_CUSTOMER_ENTRY_IDS,
+                ],
+            )
+
+            dna_samples_cases_analysis: Query = dna_samples_cases_query.join(Analysis)
```

Comment:
This join be be done in line 1635 as well it

Reply:
Changed this.